services/movie_session.py

```diff
@@ -42,3 +42,17 @@ def update_movie_session(
 
 def delete_movie_session_by_id(session_id: int) -> None:
     MovieSession.objects.get(id=session_id).delete()
+
+
+def get_taken_seats(movie_session_id: int) -> list[dict]:
+    movie_session = MovieSession.objects.get(pk=movie_session_id)
+    result = []
+    for row in range(1, movie_session.cinema_hall.rows + 1):
+        for seat in range(1, movie_session.cinema_hall.seats_in_row + 1):
+            if Ticket.objects.filter(
+                    movie_session=movie_session,
+                    row=row,
+                    seat=seat
+            ).exists():
+                result.append({"row": row, "seat": seat})
```

Comment:
wow. An advice for you. Try to avoid such constructions that make 1 sql query on each iteration(when possible)
From performance perspective it's really bad idea because if your hall has 10 rows and 10 seats, it's 100 iterations meaning 100 queries in db and it's will take too much time to just fetch available seats.
The best way is to do everything using ORM that will fetch it in 1-2 sql queries. If you work with something bigger than cinema hall

(наприклад система продажу квартир у житловому комплексі, який має 6 корпусів, у кожному по 4 під'їзди, кожен комплекс має 15 поверхів і кожен поверх має 20 квартир, то в цьому випадку ти б зробив 4 цикли(кожен корпус, під'їзд, поверх і квартира) і це було б 6*4*15*20 = 7200 запитів до бази даних, щоб подивитися які квартири вільні й де саме. Якщо записів у бд небагато, то кожен запит це 1-2 секунди. Тобто ти б витратив 7200 або 14400 секунд, або ж 2-4 години, щоб просто у додатку відобразити всі квартири. Зазвичай люди 5 секунд не можуть почекати, а тут години)

Tip: you can use .values() method to get row and seat as a dict. Everything else you should find out by yourself ;)